### PR TITLE
add lrp for gpt2

### PR DESCRIPTION
**Pull request by Tomsawyerhu**
I write a lrp-backward version for gpt2 and provide unit tests. Generally, two functions (conv1d and baddbmm) and one module(Conv1d) are newly added. Also, I modify source code of GPT2, for convenience, dropout layer are ommited.

### CODE REVIEW

**Comment by rachtibat** [comment]
Thank you for your nice work. It will take some time for me to digest your code.
We did some experiments with GPT2 (not published) and noticed that it benefits from explaining the softmax classification output with temperature scaling. Not so important now, just wanted to keep this in mind.

**Comment by rachtibat** [comment]
Update: Sorry for the delayed review! Your code looks good! I only think, we could simplify it a lot by using the `lxt.rules` instead of implementing all functions from scratch. E.g. the `Conv1DEpsilon(GPTConv1D)` is not really necessary as we can apply the `lxt.rules.EpsilonRule` on the `GPTConv1D` layer, because it is a linear operation (like we do for `nn.Linear` layers for LlaMA). https://github.com/rachtibat/LRP-eXplains-Transformers/blob/f802536a0944273377e7feca3aa4621a3b9ae23a/lxt/models/llama.py#L90

I need unfortunately more time to adapt these changes (also writing two follow-up papers right now), but I just wanted to give you a "fast" feedback and I wanted to thank you again for the really nice pull-request!

**Comment by rachtibat** [comment]
Hey,

there was actually a mistake inside the baddbmm!
I fix it the coming days and merge your pull request (:
